debug.go

```diff
@@ -20,6 +20,12 @@ func (s *spanImpl) Lock() {
 }
 
 func (s *spanImpl) maybeAssertLocked() {
+	if s.tracer == nil {
+		s.Mutex.Unlock()
```

Comment:
I guess another idea would be to defer/recover in `Lock()` and `Unlock()` if there's a panic there... might be easier to maintain.